cpp/tests/unary/math_ops_test.cpp

```diff
@@ -25,6 +25,24 @@
 
 #include <vector>
 
+using TypesToNegate =
+  cudf::test::Types<int8_t, int16_t, int32_t, int64_t, float, double, cudf::duration_D>;
+
+template <typename T>
+struct UnaryNegateTests : public cudf::test::BaseFixture {};
+
+TYPED_TEST_SUITE(UnaryNegateTests, TypesToNegate);
+
+TYPED_TEST(UnaryNegateTests, SimpleNEGATE)
+{
+  using T = TypeParam;
+  cudf::test::fixed_width_column_wrapper<T> input{{1, 2, 3}};
+  auto const v = cudf::test::make_type_param_vector<T>({-1, -2, -3});
+  cudf::test::fixed_width_column_wrapper<T> expected(v.begin(), v.end());
+  auto output = cudf::unary_operation(input, cudf::unary_operator::NEGATE);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, output->view());
+}
+
```

Comment:
We need another test that unsigned types, timestamps, strings, and other complex types fail.